fact_extractor/plugins/unpacking/maxlinear/code/maxlinear_puma7.py

```diff
@@ -0,0 +1,75 @@
+"""
+This plugin unpacks MaxLinear Puma 7 images.
+"""
+
+from pathlib import Path
+
+from common_helper_files import write_binary_to_file
+
+NAME = "MaxLinear Puma 7"
+MIME_PATTERNS = ['firmware/maxlinear-puma-7']
+VERSION = '0.1'
+
+def unpack_function(file_path, tmp_dir):
+    """
+    file_path specifies the input file.
+    tmp_dir must be used to store the extracted files.
+    Optional: Return a dict with meta information
+    """
+
+    file = Path(file_path)
+
+    try:
+        with file.open('rb') as f:
+            f.read(1) # DER
+            if f.read(1) == '\x81':
+                nbytes = 1
+            else:
+                nbytes = 2
+            der_size = int.from_bytes(f.read(nbytes), 'big')
+            f.seek(0)
+            der = f.read(der_size + 2 + nbytes)
+            outfile_der = Path(tmp_dir) / 'signature.der'
+            write_binary_to_file(der, outfile_der)
+
+            f.seek(f.tell() + f.read().find(b'$KFM'))
+            f.read(92) # skip 92 bytes
+            app_kernel_size = int.from_bytes(f.read(4), 'little')
+            f.read(928)
+            app_kernel = f.read(app_kernel_size)
+            outfile_app_kernel = Path(tmp_dir) / 'kernel.atom'
+            write_binary_to_file(app_kernel, outfile_app_kernel)
+
+            f.seek(f.tell() + f.read().find(b'$KFM'))
+            f.read(92) # skip 92 bytes
+            app_rootfs_size = int.from_bytes(f.read(4), 'little')
+            f.read(928) # skip 928 bytes
+            app_rootfs = f.read(app_rootfs_size)
+            outfile_app_rootfs = Path(tmp_dir) / 'rootfs.atom'
+            write_binary_to_file(app_rootfs, outfile_app_rootfs)
```

Comment:
These parts are very repetitive. Could you put it in a function? Maybe something like

```python
def _extract_file(fp: BufferedReader, outfile: Path):
    fp.seek(fp.tell() + fp.read().find(b'$KFM'))
    fp.read(92)  # skip 92 bytes
    file_size = int.from_bytes(fp.read(4), 'little')
    fp.read(928)  # skip 928 bytes
    content = fp.read(file_size)
    write_binary_to_file(content, outfile)
```